### PR TITLE
Better-database-config

### PR DESCRIPTION
**Pull request by SmileyChris**
This is a different take handling of DATABASE_URL, also giving the ability to have secondary configurable variables in a better way.

Also added in CACHE_URL which works the same way, along with a working CacheDsn (mostly logically copied from `django-cache-url`)

### CODE REVIEW

File: pydantic_settings/settings.py [hunk 1]
```diff
@@ -40,8 +40,13 @@ def configure(self):
             settings_dict = {
                 key: value
                 for key, value in self.settings_module().dict().items()
-                if hasattr(global_settings, key) is False
-                or value != getattr(global_settings, key)
+                if (
+                    hasattr(global_settings, key) is False
+                    # Running the test suite can modify settings.DATABASES, so always
+                    # override the mutable global_settings.DATABASES.
```
Comment (SmileyChris) [inline pydantic_settings/settings.py:58]: This took me so long to figure out what was going on :-/

File: pydantic_settings/settings.py [hunk 2]
```diff
@@ -31,8 +52,13 @@ def configure(self):
...
+                    or key == "DATABASES"
```
Comment (SmileyChris) [inline pydantic_settings/settings.py:59]: This took me so long to figure out what was going on :-/

**Comment by SmileyChris** [comment]
Hey @joshourisman , had a chance to take a look at this yet?

**Comment by joshourisman** [comment]
I have not yet, as I've been out of the country. I'm about to start working on updating an older project though, which should be a good opportunity to test this out with a real-world setup, so I'm going to try and do it this week!

**Comment by joshourisman** [comment]
Ok, this looks pretty fantastic! Definitely an improvement over some of the hacky stuff I was doing before. Tests are all passing (And better written! Thanks!), and the documentation updates looks good, so I’m going to merge!

**Pull request merged**